processor/patch.go

```diff
@@ -0,0 +1,70 @@
+package processor
+
+import (
+	"bytes"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+)
+
+type patch struct {
+	pos   int
+	stmts []ast.Stmt
+}
+
+func (p *Processor) patchFile(fset *token.FileSet, file *ast.File, patches ...patch) error {
+	src, err := formatNode(fset, file)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+
+	offset := -int(file.FileStart) + 1
+	for _, patch := range patches {
+		buf.Reset()
+
+		buf.WriteString("\n")
+		if err := format.Node(&buf, fset, patch.stmts); err != nil {
+			return err
+		}
+		buf.WriteString("\n")
+
+		pos := patch.pos + offset
```

Comment:
so this is position in file?
```suggestion
		posInFile := patch.pos - offset
```